src/Compilers/CSharp/Test/Semantic/Semantics/ReadOnlyStructsTests.cs

```diff
@@ -328,8 +592,203 @@ static void Main()
 ";
 
             var verifier = CompileAndVerify(csharp, expectedOutput: "123");
-            // PROTOTYPE: should warn about copying 'this' when calling M
-            verifier.VerifyDiagnostics();
+            verifier.VerifyDiagnostics(
+                // (11,13): warning CS8655: Call to non-readonly member 'M' from a 'readonly' member results in an implicit copy of 'this'.
+                //             M();
+                Diagnostic(ErrorCode.WRN_ImplicitCopyInReadOnlyMember, "M").WithArguments("M", "this").WithLocation(11, 13));
+        }
+
+        [Fact]
+        public void ReadOnlyAccessor_CallNormalGetAccessor()
+        {
+            var csharp = @"
+public struct S
+{
+    public int i;
+
+    public int P1
+    {
+        readonly get
+        {
+            // should create local copy
+            _ = P2; // warning
+            System.Console.Write(i);
+
+            // explicit local copy, no warning
+            var copy = this;
+            _ = copy.P2; // ok
+            System.Console.Write(copy.i);
+
+            return i;
+        }
+    }
+
+    int P2 => i = 23;
+
+    static void Main()
+    {
+        var s = new S { i = 1 };
+        _ = s.P1;
+    }
+}
+";
+
+            var verifier = CompileAndVerify(csharp, expectedOutput: "123");
+            verifier.VerifyDiagnostics(
+                // (11,17): warning CS8655: Call to non-readonly member 'get_P2' from a 'readonly' member results in an implicit copy of 'this'.
+                //             _ = P2;
+                Diagnostic(ErrorCode.WRN_ImplicitCopyInReadOnlyMember, "P2").WithArguments("get_P2", "this").WithLocation(11, 17));
+        }
+
+        [Fact]
+        public void ReadOnlyMethod_CallSetAccessor()
+        {
+            var csharp = @"
+public class C
+{
+    public int P { get; set; }
+}
+
+public struct S
+{
+    C c;
+    public S(C c)
+    {
+        this.c = c;
+        P2 = 0;
+    }
+
+    static int P1 { get; set; }
+    int P2 { get; set; }
+    readonly int P3
+    {
+        get => 42;
+        set {}
+    }
+
+    public readonly void M()
+    {
+        P1 = 1; // ok
+        P2 = 2; // error
+        P3 = 2; // ok
```

Comment:
This may have been LDM's decision, but this is my official objection to that decision. A `readonly setter` is an oxymoron.